### PR TITLE
Git add learngitbranching

### PR DESCRIPTION
**Pull request by bsipocz**
### Summary

Adds https://learngitbranching.js.org/ to the git section. We have found it to be an extremely useful resource to onboard people working with branches, pull requests, etc.

However, I'm uncertain that this is the right place to refer too it, so I keep the PR small and modular. 
I feel the changes to to `.gitignore` is uncontroversial, thus added to this PR, but I'm happy too factor it out to a separate one.

### List of changes proposed in this PR (pull-request)

* adding subsection for online git tutorial
* updating top level gitignore

### What should a reviewer concentrate their feedback on?

- [ ] Everything looks ok?

### Acknowledging contributors

- [ ] All contributors to this pull request are already named in the [table of contributors](https://github.com/alan-turing-institute/the-turing-way/blob/master/README.md#contributors) in the README file.
- [ ] The following people should be added to the [table of contributors](https://github.com/alan-turing-institute/the-turing-way/blob/master/README.md#contributors) in the README file:

### CODE REVIEW

**Review by inwaves — APPROVED** [review]

**Comment by bsipocz** [comment]
This got the approval and review once before, and the merge commit should not change it (as opposed to the opinion of the triage bot). I go ahead and merge it now, to avoid any future conflicts.

**Pull request merged**
